HW02_A_Graph-Fails.Rmd

```diff
@@ -1,87 +1,146 @@
-  theme(legend.direction = "horizontal") + 
-  theme(legend.position = c(5, 40))
-
-```
-
-### Graph Fail 4
-I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
-
-```{r}
-mpg %>% 
-ggplot(mapping = aes(x = displ, y = hwy, color = drv)) + 
-  geom_point() + 
-  geom_smooth(se = F) #se = F makes it so it won't show the error in the line of fit
-```
-
-### Graph Fail 5
-I got tired of the points, so I went to boxplots instead. However, I wanted the boxes to be all one color, but setting the color aesthetic just changed the outline? How can I make the box one color, not just the outline? 
-
-Also, the x-axis labels were overlaping, so I rotated them. But now they overlap the bottom of the graph. How can I fix this so axis labels aren't on the graph?
-
-
-```{r}
-ggplot(data = mpg, mapping = aes(x = manufacturer, y = cty, color = manufacturer)) + 
-  geom_boxplot() + 
-  theme(axis.text.x = element_text(angle = 45))
-```
-
-
-
+---
+title: "What went wrong?"
+author: "Monty Kwan edits"
+date due: "7/13/2020"
+output: github_document
+---
+
+```{r setup, include=TRUE}
+knitr::opts_chunk$set(echo = TRUE, error = TRUE)
+```
+
+## HW02 Part A
+
+In this document, I will add some examples of some coding mistakes, it is up to you to figure out why the graphs are messing up. 
+
+### First load packages
+
+It is always best to load the packages you need at the top of a script. It's another common coding formatting standard (like using the assignment operator instead of the equals sign). In this case, it helps people realize what they need to install for the script and gives an idea of what functions will be called. 
+
+It is also best coding practice to only call the packages you use, so if you use a package but end up tossing the code you use for it, then make sure to remove loading it in the first place. For example, I could use `library("tidyverse")` but since this script will only be using ggplot2, I only load ggplot2.
+
+```{r load libraries}
+library("ggplot2")
+library("magrittr") #so I can do some piping
+```
+
+
+### Graph Fail 1
+
+What error is being thrown? How do you correct it? (hint, the error message tells you)
+
+```{r}
+data(mpg) #this is a dataset from the ggplot2 package
+
+mpg %>% 
+  ggplot(mapping = aes(x = city, y = hwy, color = "blue")) %>% 
+  geom_point()
+
+```
+**My answer: This code _pipes_ the first layer of the graph into the 'geom_point' layer, which is wrong. Instead, we need to use '+' to _add_ the 'geom_point' layer to the graph. Also, look at the data columns in data(mpg), and 'city' is spelled 'cty' so this needs to be adjusted in the aes(x=cty). Are _data_ = mpg and _mapping_ = aes necessary in the code??**
+
+#### My corrected graph 1:
+
+```{r}
+data(mpg) #this is a dataset from the ggplot2 package
+
+ggplot(mpg, aes(x = cty, y = hwy, color = "blue")) +
```

Comment:
Omg, just realized this is your answer, not the previous ones....
Yes, this is correct!!